### ISSUE TITLE
[BUG][miopenBatchNormalizationForwardTraining][GFX1011] MIOpen fails to compile inline asm for GFX1011

### ISSUE THREAD

**Issue by densamoilov** [ROCm/MIOpen#1586]
Running `miopenBatchNormalizationForwardTraining` for `miopenBNSpatial` on `GFX1011` results in using `gcn_reduce2` which is implemented via inline asm that doesn't seem to be supported by `GFX1011`. I found that you recently [disabled ](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/62d20d44f34c6539f423ece93a709b71614644fa/src/kernels/MIOpenBatchNormFwdTrainSpatial.cl#L36)`gcn_reduce2` for `GFX103X` but it seems that `GFX1011` should have also been included.

Error output:
```bash
error: cannot compile inline asm
<inline asm>:17:20: error: not a valid operand.
v_add_f32 v2 v2 v2 row_bcast: 31 row_mask: 0xc
                               ^
error: cannot compile inline asm
<inline asm>:18:20: error: not a valid operand.
v_add_f32 v1 v1 v1 row_bcast: 31 row_mask: 0xc
                               ^
error: cannot compile inline asm
4 errors generated.
fwd_args->mean_0x7feaa003c0000x55bf615f3c14MIOpen Error: /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/MIOpen/src/tmp_
dir.pp:45: Can't execute cd /tmp/miopen-MIOpenBatchNormFwdTrainSpatial.cl-5332-8fcb-f041-3039;/opt/rocm-5.0.0/bin/clang-ocl -DMIOPEN_USE.
FP16=0 -DMIOPEN_USE_FP32=1 -DMIOPEN_USE_FPMIX=0 -DMIO_SAVE_MEAN_VARIANCE=1 -DMIO_RUNNING_RESULT=1 -DMIO BN_VARIANT=0 -DMIO_BN_LDS_SIZE=256
-DMIO_BN_LDSGCN_SIZE=4 -DMIO_BN_N=1 -DMIO_BN_GRP1=256 -DMIO_BN_GRP1=1 -DMIO_ BN_GRP2=1 -DMIO_ BN_GFX1030=0 -DMIO_BN_C=16 -DMIO_BN_HW=1 -DMIO_BN_NHW=1 -DMIO_BN_CHW=16 -DMIO_BN_NCHW=16 -mcpu=gfx1011 -Wno-everything -mcode-object-version=4 MIOpenBatchNormFwdTrainSpatial.cl -o /tmp/m
iopen-MIOpenBatchNormFwdTrainSpatial.cl-5332-8fcb-f041-3039/MIOpenBatchNormFwdTrainSpatial.cl.o
terminate called after throwing an instance of 'dnnl::impl::gpu::amd::miopen_error'
what(): At:/home/ms1_user/nikhil/newrepo/oneDNN/src/gpu/amd/miopen_batch_normalization_impl.hpp:22@miopenBatchNormalizationForwardTraining:miopenStatusUnknownError
Aborted (core dumped)
```
ROCm: 5.0.0
MIOpen 2.16

**Comment by junliume** [comment]
GFX1011 (Navi12) is not officially supported and legacy system. However, it seems to be a straightforward change we can make and let's discuss it @muralinr

**Comment by densamoilov** [comment]
@junliume, thanks for the quick response. Is there a list of officially supported architectures?

**Comment by junliume** [comment]
@densamoilov it is listed in the ROCm installation guide:
https://docs.amd.com/bundle/ROCm-Installation-Guide-v5.1.3/page/Prerequisite_Actions.html#d3019e198

**Comment by daniellowell** [comment]
@junliume No gfx10xx will support broadcasting down with DPP instructions as SIMD width 32 is not compatible with this type of instruction. Currently MIOpen filters via gfx1030, but that is not sufficient and will encounter this issue with any architecture gfx10xx or greater.

**Comment by atamazov** [comment]
@junliume 
> However, it seems to be a straightforward change...

Unfortunately, no. There are some important differences between Navi2x (RDNA2, Radeon RX 6000) and Navi1x (RDNA1, Radeon RX5000). IIRC Navi1x lacks dot product ALU ops that Navi2x have, and some Navi2x FMA operations are not available on Navi1x (they have MAD instead, but MAD has worse precision). So we have to check these aspects at least.

And I think we can remove the `bug` label.

@densamoilov Please indicate importance of Navi1x support for you.

**Comment by atamazov** [comment]
@daniellowell Yep, it seems like it is worth to filter out DPP for all gfx10 devices.

**Comment by densamoilov** [comment]
> Please indicate importance of Navi1x support for you.

@atamazov, not critical, we just happen to use it for testing. 

I think, in documentation we will refer to the list of officially supported hardware in the case oneDNN is used with MIOpen so as long as it's not there I think we are good.

**Comment by atamazov** [comment]
@densamoilov All right, thanks. I think we shall track this ticket as a low-priority enhancement (if everyone agrees)

**Comment by muralinr** [comment]
> @densamoilov All right, thanks. I think we shall track this ticket as a low-priority enhancement (if everyone agrees)

@atamazov I agree. It's low priority enhancement.

**Comment by ppanchad-amd** [comment]
Issue is no longer relevant. Closing ticket.
